adapters/adkernel/adkernel.go

```diff
@@ -78,8 +87,8 @@ func validateImpression(imp *openrtb2.Imp, impExt *openrtb_ext.ExtImpAdkernel) e
 	if impExt.ZoneId < 1 {
 		return newBadInputError(fmt.Sprintf("Invalid zoneId value: %d. Ignoring imp id=%s", impExt.ZoneId, imp.ID))
 	}
-	if imp.Video == nil && imp.Banner == nil && imp.Native == nil {
-		return newBadInputError(fmt.Sprintf("Invalid imp id=%s. Expected imp.banner / imp.video / imp.native", imp.ID))
+	if imp.Video == nil && imp.Banner == nil && imp.Native == nil && imp.Audio == nil {
+		return newBadInputError(fmt.Sprintf("Invalid imp id=%s. Expected imp.banner / imp.video / imp.native / imp.audio", imp.ID))
```

Comment:
you do not need this check here. This is already being checked by the [core codebase.](https://github.com/prebid/prebid-server/blob/ec729e6e5d50a1306d76332fda52108b9d313c1d/endpoints/openrtb2/auction.go#L1047)